templates/ui-frameworks/vue/collection/ui/src/{{dna_role_name}}/{{coordinator_zome_manifest.name}}/{{pascal_case collection_name}}.vue.hbs

```diff
@@ -15,73 +15,53 @@
+
+const hashes = ref<Array<ActionHash> | undefined>(undefined);
+const loading = ref(false);
+const error = ref<HolochainError | undefined>(undefined);
 
-export default defineComponent({
-  components: {
-    {{pascal_case referenceable.name}}Detail,
-  },
 {{#if (eq collection_type.type "ByAuthor")}}
-  props: {
-    author: {
-      type: Object,
-      required: true
-    }
-  },
+const props = defineProps<{author: AgentPubKey}>();
 {{/if}}
-  data(): { hashes: Array<{{referenceable.hash_type}}> | undefined; loading: boolean; error: any } {
-    return {
-      hashes: undefined,
-      loading: false,
-      error: undefined
-    }
-  },
-  async mounted() {
-  {{#if (eq collection_type.type "ByAuthor")}}
-    if (!this.author) {
-      throw new Error(`The author property is required for the {{pascal_case collection_name}} element`);
-    }
 
-  {{/if}}
-    await this.fetch{{pascal_case referenceable.name}}();
-    toRaw(this.client)?.on('signal', signal => {
-      if (signal.type !== "app") return;
-      if (signal.value.zome_name !== '{{coordinator_zome_manifest.name}}') return;
-      const payload = signal.value.payload as {{pascal_case coordinator_zome_manifest.name}}Signal;
-      if (payload.type !== 'EntryCreated') return;
-      if (payload.app_entry.type !== '{{pascal_case referenceable.name}}') return;
-    {{#if (eq collection_type.type "ByAuthor")}}
-      if (this.author.toString() !== this.client.myPubKey.toString()) return;
-    {{/if}}
-      if (this.hashes) this.hashes.push({{#if (eq referenceable.hash_type "ActionHash")}}payload.action.hashed.hash{{else}}(payload.action.hashed.content as NewEntryAction).entry_hash{{/if}});
+const { client } = inject(ClientKey)!;
+
+const fetch{{pascal_case referenceable.name}} = async () => {
+  try {
+    loading.value = true;
+    const links: Array<Link> = await client.value?.callZome({
+      role_name: '{{dna_role_name}}',
+      zome_name: '{{snake_case coordinator_zome_manifest.name}}',
+      fn_name: 'get_{{snake_case collection_name}}',
+      {{#if (eq collection_type.type "ByAuthor")}}payload: props.author,{{/if}}
     });
-  },
-  methods: {
-    async fetch{{pascal_case referenceable.name}}() {
-      try {
-        this.loading = true;
-        const links: Array<Link> = await this.client.callZome({
-          role_name: '{{dna_role_name}}',
-          zome_name: '{{snake_case coordinator_zome_manifest.name}}',
-          fn_name: 'get_{{snake_case collection_name}}',
-          {{#if (eq collection_type.type "ByAuthor")}}payload: this.author,{{/if}}
-        });
-        this.hashes = links.map(l => l.target);
-      } catch (e) {
-        this.error = e as HolochainError;
-      } finally {
-        this.loading = false;
-      }
-    }
-  },
-  setup() {
-    const client = (inject('client') as ComputedRef<AppClient>).value;
-    return { client };
-  },
+    hashes.value = links.map(l => l.target);
+  } catch(e) {
+    error.value = e as HolochainError;
```

Comment:
You may as well set `useUnknownInCatchVariables: false` in the `tsconfig.app.json` and omit the type assertion.